arraycontext/container/dataclass.py

```diff
@@ -127,36 +136,52 @@ def is_array_field(f: Field, field_type: type) -> bool:
 
         return is_array_type(field_type)
 
+    from pytools import partition
+
+    array_fields = _get_annotated_fields(cls)
+    array_fields, non_array_fields = partition(is_array_field, array_fields)
+
+    if not array_fields:
+        raise ValueError(f"'{cls}' must have fields with array container type "
+                "in order to use the 'dataclass_array_container' decorator")
+
+    return _inject_dataclass_serialization(cls, array_fields, non_array_fields)
+
+
+def _get_annotated_fields(cls: type) -> Sequence[_Field]:
```

Comment:
This could be a `_partition_fields(cls, predicate)` instead?

Reply:
I'm OK with the way it is.